pyomo/core/base/units_container.py

```diff
@@ -1013,13 +1049,27 @@ def exitNode(self, node, data):
         # first check if the node is a leaf
         if type(node) in nonpyomo_leaf_types \
                 or not node.is_expression_type():
-            if isinstance(node, _PyomoUnit):
+            if type(node) in native_numeric_types:
+                # this is a number - return dimensionless                                                                      
+                return (None, None)
```

Comment:
This if-block is not necessary - it will get caught by the fall-through below - but maybe this is better for performance...

Reply:
Yes, because most trees have a lot of native numeric leaf nodes, and `isinstance` is quite slow when it returns False.